parser/src/lexer.rs

```diff
@@ -315,10 +315,21 @@ where
 
             // 1e6 for example:
             if self.chr0 == Some('e') || self.chr0 == Some('E') {
+                if self.chr1 == Some('_') {
```

Comment:
These are identical from what I can tell. Probably a good idea to refactor into a function (`check_invalid_underscore`?) that returns the error for you (which for tokenizing a number has the message `'invalid decimal literal'` if I'm not mistaken).

Reply:
@leesungbin will you check @DimitrisJim 's suggestion?